### ISSUE TITLE
definition components usage and missing information

### ISSUE THREAD

**Issue by sadhasivam** [coveooss/json-schema-for-humans#190]
Our schema has "$def" section where we have created reusable components. We reference the specified "$def" elements across multiple places with additional information. 

The generator also references the "$def". However, schema docs need contextual values added. Any thoughts?

## Sample Schema

```json
{
  "$schema": "https://json-schema.org/draft/2019-09/schema",
  "type": "object",
  "title": "OF",
  "properties": {
    "uuid": {
      "$ref": "#/$defs/ofString",
      "description": "Unique Identifer",
      "examples": [
        "29292929292929292929292"
      ]
    },
    "firstName": {
      "$ref": "#/$defs/ofString",
      "description": "first name",
      "minLength": 10,
      "examples": [
        "John"
      ]
    },
    "lastName": {
      "$ref": "#/$defs/ofString",
      "description": "last name",
      "maxLength": 10,
      "examples": [
        "Doe"
      ]
    }
  },
  "$defs": {
    "ofString": {
      "type": "string",
      "minLength": 5,
      "maxLength": 250
    }
  }
}

```

## Output 

<img width="622" alt="image" src="https://user-images.githubusercontent.com/253447/225800306-dfd82eda-4cc4-48b3-9644-ed29c2936d93.png">

## Issue

# localized firstName, lastName examples , constraints are missing.

**Comment by gurland** [comment]
Same here

**Comment by dblanchette** [comment]
Should be fixed in version 0.48, let me know if not
